apps/balancer-tools/src/app/poolsimulator/(utils)/index.ts

```diff
@@ -26,45 +51,39 @@ export function convertAnalysisDataToAMM(data: AnalysisData) {
             priceRate: String(token.rate),
           })),
           tokensList: data.tokens.map((token) => String(token.symbol)),
-        }),
+        })
       );
     }
     case PoolTypeEnum.GyroE: {
-      return new AMM(
-        new ExtendedGyroEV2({
-          swapFee: String(data.poolParams?.swapFee),
-          totalShares: String(
-            data.tokens.reduce((acc, token) => acc + token.balance, 0),
-          ),
-          tokens: data.tokens.map((token) => ({
-            address: String(token.symbol), // math use address as key, but we will use symbol because custom token will not have address
-            balance: String(token.balance),
-            decimals: token.decimal,
-            priceRate: String(token.rate),
-          })),
-          tokensList: data.tokens.map((token) => String(token.symbol)),
+      return fetchECLPDerivativeParams(data)
+        .then((derivedParams) => {
+          return new AMM(
+            new ExtendedGyroEV2({
+              swapFee: String(data.poolParams?.swapFee),
+              totalShares: String(
+                data.tokens.reduce((acc, token) => acc + token.balance, 0)
+              ),
+              tokens: data.tokens.map((token) => ({
+                address: String(token.symbol), // math use address as key, but we will use symbol because custom token will not have address
+                balance: String(token.balance),
+                decimals: token.decimal,
+                priceRate: String(token.rate),
+              })),
+              tokensList: data.tokens.map((token) => String(token.symbol)),
 
-          gyroEParams: {
-            alpha: String(data.poolParams?.alpha),
-            beta: String(data.poolParams?.beta),
-            lambda: String(data.poolParams?.lambda),
-            c: String(data.poolParams?.c),
-            s: String(data.poolParams?.s),
-          },
-          derivedGyroEParams: {
-            tauAlphaX: String(data.poolParams?.tauAlphaX),
-            tauAlphaY: String(data.poolParams?.tauAlphaY),
-            tauBetaX: String(data.poolParams?.tauBetaX),
-            tauBetaY: String(data.poolParams?.tauBetaY),
-            u: String(data.poolParams?.u),
-            v: String(data.poolParams?.v),
-            w: String(data.poolParams?.w),
-            z: String(data.poolParams?.z),
-            dSq: String(data.poolParams?.dSq),
-          },
-          tokenRates: data.tokens.map((token) => String(token.rate)),
-        }),
-      );
+              gyroEParams: {
+                alpha: String(data.poolParams?.alpha),
+                beta: String(data.poolParams?.beta),
+                lambda: String(data.poolParams?.lambda),
+                c: String(data.poolParams?.c),
+                s: String(data.poolParams?.s),
+              },
```

Comment:
You're doing the same thing above, a potential refactor here

Reply:
Is almost the same, here we don't rename the `lambda`